source/slang/slang-emit-hlsl.cpp

```diff
@@ -500,6 +500,11 @@ bool HLSLSourceEmitter::tryEmitInstExprImpl(IRInst* inst, const EmitOpInfo& inOu
 {
     switch (inst->getOp())
     {
+        case kIROp_ControlBarrier:
+        {
+            m_writer->emit("GroupMemoryBarrier();\n");
```

Comment:
GroupMemoryBatrierWithGroupSync.

Otherwise this is just a memory barrier not a control barrier.